packages/Toggle/src/Toggle.js

```diff
@@ -1,7 +1,352 @@
+  position: absolute;
+  top: -2px;
+  bottom: -2px;
+  left: -2px;
+  right: -2px;
+  border: 2px solid rgba(67, 177, 229, 0.25);
+  border-radius: 50px;
+  opacity: 0;
+  transform: scale(0.8);
+
+  ${toggleInput.selector}:focus ~ & {
+    opacity: 1;
+    transform: scale(1);
+  }
+`;
+
+const labelStyleBase = `
+  transition: all ${transitionInMS}ms ease-in-out;
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  margin: auto;
+  height: 11px;
+  line-height: 11px;
+  text-transform: uppercase;
+  font-weight: bold;
+  font-size: 11px;
+  color: white;
+  user-select: none;
+`;
+
+const onLabelStyle = css`
+  ${labelStyleBase};
+  left: 9px;
+  color: #bbebff;
+
+  ${toggleInput.selector}:hover ~ ${toggleGroove.selector} > &,
+  ${toggleInput.selector}:focus ~ ${toggleGroove.selector} > & {
+    color: white;
+  }
+`;
+
+const offLabelStyle = css`
+  ${labelStyleBase};
+  right: 6px;
+  color: #9fa1a2;
+`;
+
+const getStatefulContainerStyles = ({ size, disabled }) => {
+  const baseStyle = css`
+    position: relative;
+    display: inline-block;
+    cursor: ${disabled ? 'not-allowed' : 'pointer'};
+  `;
+
+  const sizeStyle = {
+    default: css`
+      height: 32px;
+      width: 62px;
+    `,
+
+    small: css`
+      height: 22px;
+      width: 40px;
+    `,
+
+    xsmall: css`
+      height: 14px;
+      width: 26px;
+    `,
+  };
+
+  return css`
+    ${baseStyle} ${sizeStyle[size]}
+  `;
+};
+
+const getStatefulGrooveStyles = ({ variant, checked, disabled }) => {
+  const baseStyles = `
+    transition: ${transitionInMS}ms all ease-in-out, 0 background-color linear;
+    display: inline-block;
+    overflow: hidden;
+    flex-shrink: 0;
+    position: relative;
+    border-radius: 50px;
+    overflow: hidden;
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    border: 1px solid;
+
+    // We're animating this pseudo-element in order to give the toggle groove
+    // background an animation in and out.
+    &:before {
+      content: '';
+      transition: ${transitionInMS}ms all ease-in-out;
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      border-radius: 50px;
+      background-color: ${colors.mongodb.blue};
+      box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.1);
+      opacity: 0;
+      transform: scale(0.85);
+    }
+
+    ${
+      disabled
+        ? `
+      &:before {
+        opacity: 0;
+      }
+    `
+        : ''
+    }
+
+    ${
+      checked && !disabled
+        ? `
+      // We set background-color here to avoid a small issue with overflow clipping
+      // that makes this look less seamless than it should.
+      background-color: ${colors.mongodb.blue};
+      border-color: #2e9ed3;
+      transition-delay: ${transitionInMS}ms;
+
+      &:before {
+        transform: scale(1);
+        opacity: 1;
+      }
+    `
+        : ''
+    }
+  `;
+
+  const variants = {
+    default: css`
+      ${disabled
```

Comment:
We have these variables in state...all of these styles can be discrete css`` calls that we customize on the component itself. 

Right now I find this very hard to piece together what the underlying style is. 

We could have 

```
const variants = {
  default: ({ disabled, checked }) => {
    if (disabled) return css``
  }
}
```

Similar to what's in Checkbox.js.

Reply:
I'm not so sure about this. I will say that the ternary expressions made the styles hard to read. I like having the styles for each element in one place. Take a look at the updates I made. I find it a bit more reasonable to parse through.

If that's still not working, I'll take this for a spin. I'm just worried about having so many places where we will need to pass the same 3 parameters in (`checked, disabled, variant`) and (`checked, disabled, size`) since there's a lot of styles that depend on at least two states.